### PR TITLE
Only send intermediate chain if specified

### PR DESCRIPTION
**Pull request by psytale**

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+build:
+  image: docker:1.7.1
+  volumes:
+    - /var/run/docker.sock:/var/run/docker.sock
+  commands:
+    - mkdir -p /etc/docker/certs.d/docker-registry.expend.io/
+    - wget https://s3-eu-west-1.amazonaws.com/io.expend.production.certificates/curoo-root.pem -O /etc/docker/certs.d/docker-registry.expend.io/ca.crt
```
Comment (rymir) [inline .drone.yml]: curoo-root.pem is public right?

Comment (psytale) [inline .drone.yml]: Yeah the whole bucket is

File: .drone.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+build:
+  image: docker:1.7.1
```
Comment (rymir) [inline .drone.yml]: isn't it better to have a `build_haproxy` where we encapsulate what we have now in `commands`?

Comment (psytale) [inline .drone.yml]: could probably just use build_base come to think of it, just copied expend-docker-build and didn't think of it again.

File: bootstrap.py [hunk 3]
```diff
@@ -31,29 +39,36 @@ def get_object(file):
...
-parser.read('/bootstrap/config.cfg')
-passphrase = parser['SSL']['rsa_passphrase']
-if 'client_validation' in parser['HAPROXY']:
-    with open('/bootstrap/client-certificate-chain.pem', 'wb') as chain:
-        chain.write(get_object('client-certificate-chain.pem'))
+    os.mkdir('/bootstrap')
 
-# haproxy doesn't support key with passprhase so remove it
-subprocess.call(['openssl rsa -in /bootstrap/key.pem -passin pass:' + passphrase + ' -out /bootstrap/key.pem'], shell=True)
+config = yaml.load(get_object(config))
```
Comment (rymir) [inline bootstrap.py]: I would rename `get_object` to `get_s3_object`, otherwise when I see it here I have no clue what object we are talking about

Comment (psytale) [inline bootstrap.py]: My plan is that eventually it could deal with other urls. i.e. if it is file:// it opens it from the local filesystem instead.

Comment (psytale) [inline bootstrap.py]: I think a general get function makes a lot more sense, especially for local testing, can have a config with file://local-path/config.yml and still use the same functions

Comment (rymir) [inline bootstrap.py]: ok

Comment (psytale) [inline bootstrap.py]: How about get_file_from_url?

Comment (psytale) [inline bootstrap.py]: read_file_from_url would make more sense with what it is doing.

Comment (rymir) [inline bootstrap.py]: good for me

File: Dockerfile [hunk 4]
```diff
@@ -2,12 +2,13 @@ FROM haproxy:1.5
 
 RUN apt-get update && \
     apt-get install -y \
+        git \
         openssl \
         python3 && \
     apt-get clean
 ADD https://bootstrap.pypa.io/get-pip.py /get-pip.py
 RUN python3 get-pip.py && rm /get-pip.py
```
Comment (rymir) [inline Dockerfile]: why not adding `python3-pip` to `apt-get install`?

File: Dockerfile [hunk 5]
```diff
@@ -2,12 +2,13 @@ FROM haproxy:1.5
...
-RUN pip install boto3
+RUN pip install pyyaml==3.11 git+git://github.com/curoo/expend-python-commons.git@master#egg=ex_py_commons
```
Comment (rymir) [inline Dockerfile]: we should probably add this `expend-python-commons` to our `deploy_base` too in https://github.com/curoo/expend-docker-build/blob/445f919ebd26745aa8626e7624fd62f7c138fb62/deploy_base/Dockerfile#L12

File: Dockerfile (hunk 4 above)
Comment (psytale) [inline Dockerfile]: pretty sure the distro this image was running didn't have that available.

File: Dockerfile (hunk 5 above)
Comment (psytale) [inline Dockerfile]: Yeah agreed, but that isn't relevant to this PR since we aren't building from it

**Pull request merged**